packages/material-ui/src/Box/styleFunction.test.js

```diff
@@ -213,5 +213,27 @@ describe('styleFunction', () => {
         '@media (min-width:960px)': { padding: '70px', margin: '30px' },
       });
     });
+
+    it('writes breakpoints in correct order', () => {
+      const result = styleFunction({
+        theme,
+        m: { md: 1, lg: 2 },
+        sx: { p: { xs: 0, sm: 1, md: 2 } },
+      });
+
+      expect(Object.keys(result)).to.deep.equal([
```

Comment:
deep equal test the structure in both directions, to avoid confusion

```suggestion
      // Test the order
      expect(Object.keys(result)).to.deep.equal([
```